remill/Arch/Runtime/Operators.h

```diff
@@ -465,6 +465,27 @@ MAKE_WRITE_REF(float64_t)
 
 #undef MAKE_WRITE_REF
 
+#define MAKE_CMPXCHG(size, type_prefix, access_suffix) \
+    ALWAYS_INLINE static bool _CmpXchg( \
+        Memory *&memory, RnW<type_prefix ## size ## _t> op, \
+        type_prefix ## size ## _t expected, type_prefix ## size ## _t desired) { \
+      return __remill_compare_exchange_ ## access_suffix (memory, op.val_ref, &expected, desired);\
```

Comment:
Here I think you should just implement the semantics of the non-memory compare exchange.